### PR TITLE
Update allowed C++ version

### PR DESCRIPTION
**Pull request by asfernandes**

### CODE REVIEW

File: doc/README.modern_cpp.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
 # Modern C++ features
 
-We define here as "modern C++" features the ones introduced since C++17. These features should be used in a "controlled" form in Firebird code, as not all compilers fully support them.
+It's allowed to use in Firebird internal code C++ features up to and including C++17.
 
-Only ones mentioned in this document could be used, but as necessities appears, discussion should be started in the devel list or pull requests.
+Public files (as API headers) is limited to C++11.
```
Comment (asfernandes) [inline doc/README.modern_cpp.md:5]: Note that I also want to change version of public code to C++11.

**Comment by asfernandes** [comment]
If we agree on C++17, I will also update build files.

File: doc/README.modern_cpp.md (hunk 1 above)
Comment (hvlad) [reply doc/README.modern_cpp.md:5]: How this limitation can be enforced on practice ?

**Comment by hvlad** [comment]
> If we agree on C++17, I will also update build files.

No objections.

File: doc/README.modern_cpp.md (hunk 1 above)
Comment (asfernandes) [reply doc/README.modern_cpp.md:5]: We may build the examples with this option.

**Comment by asfernandes** [comment]
> If we agree on C++17, I will also update build files.

As we had previously defined C++14 with C++17 extensions, build files are already defined to C++17.

File: doc/README.modern_cpp.md (hunk 1 above)
Comment (aafemt) [reply doc/README.modern_cpp.md:5]: So now API headers can handle std::exception?

Comment (asfernandes) [reply doc/README.modern_cpp.md:5]: `std::exception` is pre-c++11 feature, so explain what you want in fb-devel, as it seems to have nothing related with this discussion.

**Comment by hvlad** [comment]
Looks like some incompatibility (with commit b31d37b, probalby) after merge: MacOS build of examples can't find path for libstdc++ headers

**Pull request merged**